addons/sale_timesheet/models/project.py

```diff
@@ -146,6 +146,20 @@ def _compute_partner_id(self):
                 sol = project.sale_line_id or project.sale_line_employee_ids.sale_line_id[:1]
                 project.partner_id = sol.order_partner_id
 
+    @api.depends('partner_id')
+    def _compute_sale_line_id(self):
+        super()._compute_sale_line_id()
+        for project in self.filtered(lambda p: not p.sale_line_id and p.partner_id and p.pricing_type == 'employee_rate'):
+            # Give a SOL by default either the last SOL with service product and remaining_hours > 0
+            sol = self.env['sale.order.line'].search([
```

Comment:
Could we avoid the search in the for loop by using a group_by and an array_agg on the SOL id ?

Reply:
It's a good idea but when I do a read_group with group_by order_partner_id to have an array of SOL id. After that, how can I know which order_partner_id in the read_group satisfies this condition : `('order_partner_id', 'child_of', project.partner_id.commercial_partner_id.id)` in the for loop ? I think I will do more read than my actual code 😬 
I don't change the code for this reason. Sorry !